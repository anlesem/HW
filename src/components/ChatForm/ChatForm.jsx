import { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

import style from './ChatForm.module.scss';

import Button from '@mui/material/Button';
import SendIcon from '@mui/icons-material/Send';
import TextField from '@mui/material/TextField';

export const ChatForm = ({ data }) => {
  const [text, setText] = useState('');

  const handleSubmit = (event) => {
    event.preventDefault();
    if (text) {
      data.changeNameChat(text);
      setText('');
    }
  };

  const handleInput = (value) => {
    setText(value);
  };

  const handleKeyDown = (event) => {
    if (event.keyCode === 13) {
      event.preventDefault();
      handleSubmit(event);
    }
  };

  useEffect(() => {
    setText('');
  }, [data.checked]);

  return (
    <form className={style.rename} onSubmit={handleSubmit}>
      <TextField
        id="outlined-basic"
        name="chat"
        label="Название чата"
        variant="outlined"
        value={text}
        onChange={(event) => handleInput(event.target.value)}
        onKeyDown={(event) => handleKeyDown(event)}
        required
        inputProps={{ 'data-testid': 'chat-form-input' }}
        className={style.textField}
      />
      <Button variant="outlined" type="submit" data-testid="chat-form-button">
        <SendIcon />
      </Button>
    </form>
  );
};

ChatForm.propTypes = {
  data: PropTypes.shape({
    checked: PropTypes.array.isRequired,
    changeNameChat: PropTypes.func.isRequired
  })
};
