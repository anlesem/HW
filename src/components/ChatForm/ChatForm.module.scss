.rename {
  position: absolute;
  top: 5px;
  width: 100%;
  background-color: #fff;
  margin-bottom: 20px;

  display: flex;
  justify-content: space-between;

  button {
    margin-left: 10px;
  }
}

.textField {
  flex-grow: 1;
}
