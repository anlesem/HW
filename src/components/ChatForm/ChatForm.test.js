import { render, screen, fireEvent } from '@testing-library/react';
import { ChatForm } from './ChatForm';

// import { Provider } from 'react-redux';
// import { store } from '../../store/store';

describe('ChatForm', () => {
  it('Компонент существует', () => {
    expect(ChatForm).toBeInstanceOf(Function);
  });

  it('Снимок состояния', () => {
    const changeNameChat = jest.fn();
    const checked = [];

    const { asFragment } = render(<ChatForm data={{ changeNameChat, checked }} />);
    expect(asFragment(<ChatForm />)).toMatchSnapshot();
  });

  it('Начало: Основное поле ввода пустое', () => {
    const changeNameChat = jest.fn();
    const checked = [];
    render(<ChatForm data={{ changeNameChat, checked }} />);

    const input = screen.getByTestId('chat-form-input');
    expect(input.value).toBe('');
  });

  it('Заполнение поле ввода', () => {
    const changeNameChat = jest.fn();
    const checked = [];
    render(<ChatForm data={{ changeNameChat, checked }} />);

    const input = screen.getByTestId('chat-form-input');

    fireEvent.change(input, { target: { value: 'bla-bla-bla' } });
    expect(input.value).toBe('bla-bla-bla');
  });

  it('Обработка клика на кнопку', () => {
    const changeNameChat = jest.fn();
    const checked = [];
    render(<ChatForm data={{ changeNameChat, checked }} />);

    const input = screen.getByTestId('chat-form-input');
    const button = screen.getByTestId('chat-form-button');

    fireEvent.change(input, { target: { value: 'bla-bla-bla' } });
    expect(input.value).toBe('bla-bla-bla');

    fireEvent.click(button);
    expect(changeNameChat).toBeCalled();
    expect(input.value).toBe('');
  });
});
