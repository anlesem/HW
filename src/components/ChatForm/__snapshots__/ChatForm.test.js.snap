// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatForm Снимок состояния 1`] = `
<DocumentFragment>
  <form
    class="rename"
  >
    <div
      class="MuiFormControl-root MuiTextField-root textField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="outlined-basic"
        id="outlined-basic-label"
      >
        Название чата
        <span
          aria-hidden="true"
          class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           *
        </span>
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          data-testid="chat-form-input"
          id="outlined-basic"
          name="chat"
          required=""
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-173wfxe"
          >
            <span>
              Название чата *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      data-testid="chat-form-button"
      tabindex="0"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SendIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </form>
</DocumentFragment>
`;
